src/network/netdev/wireguard.c

```diff
@@ -304,11 +331,23 @@ static void resolve_endpoints(NetDev *netdev) {
 
                 if (r == -ENOBUFS)
                         break;
+                if (r < 0) {
+                        log_netdev_error_errno(netdev, r, "Failed to create resolver: %m");
+                        continue;
+                }
 
-                LIST_REMOVE(endpoints, w->unresolved_endpoints, endpoint);
+                r = sd_resolve_query_set_destroy_callback(q, wireguard_endpoint_destroy_callback);
+                if (r < 0) {
+                        log_netdev_error_errno(netdev, r, "Failed to set destroy callback to resolving query: %m");
+                        continue;
+                }
 
-                if (r < 0)
-                        log_netdev_error_errno(netdev, r, "Failed create resolver: %m");
+                (void) sd_resolve_query_set_floating(q, true);
+
+                /* To avoid netdev is freed. It it unrefed by the destroy callback. */
```

Comment:
typo:

```diff
-/* To avoid netdev is freed. It it unrefed by the destroy callback. */ 
+/* To avoid netdev is freed. It is unrefed by the destroy callback. */ 
```